### ISSUE TITLE
False positive recur mismatch with defrecord + letfn / let + fn

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1806]
``` clojure
$ clj-kondo --lint - <<< '(defrecord Foo [] clojure.lang.ILookup (valAt [_ _] (letfn [(foo [x] (recur 1 2))])))'
<stdin>:1:62: warning: unused binding foo
<stdin>:1:67: warning: unused binding x
<stdin>:1:70: error: recur argument count mismatch (expected 0, got 2)
linting took 6ms, errors: 1, warnings: 2
```

```
$ clj-kondo --lint - <<< '(defrecord Foo [] clojure.lang.ILookup (valAt [_ _] (let [foo (fn foo [x] (recur 1 2))])))'
<stdin>:1:59: warning: unused binding foo
<stdin>:1:72: warning: unused binding x
<stdin>:1:75: error: recur argument count mismatch (expected 1, got 2)
linting took 12ms, errors: 1, warnings: 2
```

Reported by @quoll
